### DISCUSSION TITLE
Sometimes the proxy connection doesn't work

### DISCUSSION THREAD

**Discussion by AngeloAvv** [bluenviron/mediamtx#269] [Questions]
## Which version are you using?

v0.9.12

## Describe the problem

I'm trying to proxy 8 different rtsp streams using rtsp-simple-server. Sometimes a strange behavior occurs. VLC media player starts connecting but an error is shown. Here's the log:

```
2020/09/05 12:12:43 [8/0/7] [source cam3] starting on demand
2020/09/05 12:12:43 [8/0/7] [source cam3] started
2020/09/05 12:12:43 [8/0/7] [source cam3] connecting
2020/09/05 12:12:48 [9/0/7] [client [::1]:43348] connected
2020/09/05 12:12:48 [9/0/7] [client [::1]:43348] SETUP
2020/09/05 12:12:48 [9/0/7] [client [::1]:43348] ERR: the path must contain a base path and a control path (cam3)
2020/09/05 12:12:48 [8/0/7] [client [::1]:43348] disconnected
2020/09/05 12:12:48 [8/0/7] [client [::1]:43344] ERR: publisher of path 'cam3' has timed out
2020/09/05 12:12:48 [7/0/7] [client [::1]:43344] disconnected
2020/09/05 12:12:48 [8/0/7] [client [::1]:43350] connected
2020/09/05 12:12:48 [8/0/7] [client [::1]:43350] OPTIONS
2020/09/05 12:12:48 [8/0/7] [client [::1]:43350] ERR: path must begin with a slash
2020/09/05 12:12:48 [7/0/7] [client [::1]:43350] disconnected
2020/09/05 12:12:53 [7/0/7] [source cam3] stopping since we're not requested anymore
2020/09/05 12:12:53 [7/0/7] [source cam3] stopped
```

That's weird because it occurs randomly on different streams. If I try to open the stream again 2 or 3 minutes later it works and maybe the problem moves to another proxy stream (cam5). I thought it was a performance related issue because I'm working on an Orange Pi with a Quad Core CPU and 256MB of RAM, but I excluded that because if I open more streams from other entries, it works perfectly.

Any ideas?

**Comment by aler9** [comment]
Hello, what happens is the following:
1. VLC connects to the server, the source is started on demand
2. The source does not get ready within a timespan imposed by VLC
3. VLC detects that this timespan has expired and retries to connect to the server, this time by using a simpler variant of the protocol - RTSP is implemented very badly in some cameras, and VLC tries to handle all cases - this simpler variant is not supported by rtsp-simple-server
4. VLC disconnects, the source disconnects

So: you must try to reduce the response time between the server and the source - or use another client (not vlc). Maybe the problem is that the bandwidth is limited between the server and the source - in this case, try reducing the number of sources or try reducing the stream bitrate.

[a comment by an automated account is not shown]
